pageserver/src/tenant/timeline.rs

```diff
@@ -1923,10 +1955,16 @@ impl Timeline {
         )
         .await;
 
-        // TODO: work toward making this a no-op. See this funciton's doc comment for more context.
+        // TODO: work toward making this a no-op. See this function's doc comment for more context.
         tracing::debug!("Waiting for tasks...");
         task_mgr::shutdown_tasks(None, Some(self.tenant_shard_id), Some(self.timeline_id)).await;
 
+        {
+            // Allow any remaining in-memory layers to do cleanup.
+            let mut write_guard = self.write_lock.lock().await;
+            self.layers.write().await.shutdown(&mut write_guard);
+        }
```

Comment:
```suggestion
        {
            // `InMemoryLayer`s keep the timeline Gate open.
            // => destroy them before waiting for gate closure below.
            let mut write_guard = self.write_lock.lock().await;
            self.layers.write().await.shutdown(&mut write_guard);
        }
```

Reply:
Unsure why this is needed. There will be a lockup if this is skipped.

Reply:
[ff459df](https://github.com/neondatabase/neon/pull/8229/commits/ff459dfd4ac773a68ead92c7baa3e537bc0bf2e3)